api_tests/actions/views/test_schema_response_action_list.py

```diff
@@ -0,0 +1,181 @@
+        )
+
+    def make_payload(self, schema_response, trigger):
+        return {
+            'data':
+                {
+                    'type': 'schema-response-actions',
+                    'attributes': {
+                        'trigger': trigger.db_name,
+                        'comment': "It's Party time!",
+                    },
+                    'relationships': {
+                        'target': {
+                            'data': {
+                                'id': schema_response._id,
+                                'type': 'schema-responses'
+                            }
+                        }
+                    }
+                }
+        }
+
+    @pytest.fixture()
+    def url(self, schema_response):
+        return f'/v2/schema_responses/{schema_response._id}/actions/'
+
+    def test_schema_response_action_submit(self, app, registration, schema_response, user, url):
+        assert schema_response.reviews_state == ApprovalStates.IN_PROGRESS.db_name
+        registration.add_contributor(user, 'admin')
+        assert not schema_response.pending_approvers.count()
+        payload = self.make_payload(schema_response=schema_response, trigger=SchemaResponseTriggers.SUBMIT)
+        resp = app.post_json_api(url, payload, auth=user.auth, expect_errors=True)
+        assert resp.status_code == 201
+        schema_response.refresh_from_db()
+        assert schema_response.reviews_state == ApprovalStates.UNAPPROVED.db_name
+
+    def test_schema_response_action_approve(self, app, registration, unapproved_schema_response, user, url):
+        payload = self.make_payload(schema_response=unapproved_schema_response, trigger=SchemaResponseTriggers.APPROVE)
+        resp = app.post_json_api(url, payload, auth=user.auth, expect_errors=True)
+        assert resp.status_code == 201
+        unapproved_schema_response.refresh_from_db()
+        assert unapproved_schema_response.reviews_state == ApprovalStates.APPROVED.db_name
+
+    def test_schema_response_action_accept(self, app, registration, unapproved_schema_response, user, url):
+        """
+        Accept behavior is explained in the docstring for _validate_accept_trigger function. There are 3 methods of
+        using this trigger.
+        """
+        unapproved_schema_response.pending_approvers.clear()
+        payload = self.make_payload(schema_response=unapproved_schema_response, trigger=SchemaResponseTriggers.ACCEPT)
+        resp = app.post_json_api(url, payload, auth=user.auth, expect_errors=True)
+        assert resp.status_code == 201
+        unapproved_schema_response.refresh_from_db()
+        assert unapproved_schema_response.reviews_state == ApprovalStates.APPROVED.db_name
+
+    def test_schema_response_action_moderator_reject(self, app, registration, unapproved_schema_response, user, url):
+        payload = self.make_payload(schema_response=unapproved_schema_response, trigger=SchemaResponseTriggers.MODERATOR_REJECT)
```

Comment:
yikes!! MODERATOR_REJECT should only be callable if the response is PENDING_MODERATION. This is showing off the way we consolidate our triggers on the state machine, so "MODERATOR_REJECT" and "ADMIN_REJECT" are really the same thing at different times...

Reply:
Checking the created trigger would have caught this, as it would have been ADMIN_REJECT